### ISSUE TITLE
Provide Bloomfilter(),HyperLogLog() type scalar functions that work like get_dict()

### ISSUE THREAD

**Issue by michaelritsema** [ClickHouse/ClickHouse#70812]
**Use case**

A probabilistic data structure like Bloomfilter is very useful for detection algorithms like malicious IPs. 

For example.

- Load 1 billion IPs into a table that represent bad ips.
- Load it into the Bloomfilter from the table like you would with a clickhouse dictionary
- add the filter to a where clause: like ' WHERE get_bloomfilter('bad_ips',ip_address)' 

This would use a small amount of ram and useful for a quick pre-filter of data.

While I don't know of other databases that provide it out of the box there are extensions for a few such as Postgres that provide this functionality. Because Bloomfilters are already used in the product I think I can keep this brief but am happy to provide for information.

**Comment by UnamedRus** [comment]
PR https://github.com/ClickHouse/ClickHouse/pull/79648

**Comment by michaelritsema** [comment]
@UnamedRus I don't believe #79648 is relevant to this issue. This isn't related to the IN subquery clause but a scalar like has() for arrays... for bloom/HLL type probabilistic data structures. Postgres has this similar extension:

https://github.com/citusdata/postgresql-hll

**Comment by otselnik** [comment]
Bloomfilter PR https://github.com/ClickHouse/ClickHouse/pull/101841
